### ISSUE TITLE
[BUG] `PolynomialTrendForecaster` use causes error: "found array with dim 3. Estimator expected <= 2? Use forecaster = PolynomialTrendForecaster(degree=1)"

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#2117]
### Discussed in https://github.com/alan-turing-institute/sktime/discussions/832

<div type='discussions-op-text'>

<sup>Originally posted by **ljf20122010** April 21, 2021</sup>
I Use 
**forecaster = PolynomialTrendForecaster(degree=1)  ,**
and my y_trian is 1dim(**y_train_ndim=1**)
so why report Found array with dim 3. Estimator expected <= 2?
this is my code:

#-------------------Fitting an Auto Regressive Light-GBM------------
#Setting Quantile Regression Hyper-parameter.
params = {
    'objective':'quantile'
}
#A 10 percent and 90 percent prediction interval(0.1,0.9 respectively).
quantiles = [.1, .5, .9] #Hyper-parameter "alpha" in Light GBM
#Capture forecasts for 10th/median/90th quantile, respectively.
forecasts = []
#Iterate for each quantile.
for alpha in quantiles:
    

    forecaster = get_transformed_target_forecaster(alpha,params)
    
    #Initialize ForecastingHorizon class to specify the horizon of forecast
    fh = ForecastingHorizon(y_test.index, is_relative=False)
    
    #Fit on Training data.
    forecaster.fit(y_train)
    
    #Forecast the values.
    y_pred = forecaster.predict(fh)
    
    #List of forecasts made for each quantile.
    y_pred.index.name="date"
    y_pred.name=f"predicted_sales_q_{alpha}"
    forecasts.append(y_pred)

#Append the actual data for plotting.
store1_agg_monthly.index.name = "date"
store1_agg_monthly.name = "original"
forecasts.append(store1_agg_monthly)</div>

**Comment by fkiraly** [comment]
MRE by @hagen42:

Hey, I've got the same error.
At the line with _forecaster.fit(y)_ occurs following error: **Found array with dim 3. Estimator expected <= 2.**
I reproduced this error with the sktime dataset "airline". Following an example to reproduce:

```
import lightgbm as lgb

from sktime.datasets import load_airline
from sktime.transformations.series.detrend import Deseasonalizer, Detrender
from sktime.forecasting.trend import PolynomialTrendForecaster
from sktime.forecasting.compose import (
    TransformedTargetForecaster,
    RecursiveTimeSeriesRegressionForecaster,
)


def create_forecaster_w_desesonalizer(sp, degree=1):
    regressor = lgb.LGBMRegressor()
    forecaster = TransformedTargetForecaster(
        [
            ("deseasonalize", Deseasonalizer(model="additive", sp=sp)),
            ("detrend", Detrender(forecaster=PolynomialTrendForecaster(degree=degree))),
            (
                "forecast",
                RecursiveTimeSeriesRegressionForecaster(
                    regressor,
                    window_length=96,
                ),
            ),
        ]
    )
    **return** forecaster


y = load_airline()

forecaster = create_forecaster_w_desesonalizer(sp=12)

forecaster.fit(y)

print(forecaster.summary())
```

**Comment by fkiraly** [comment]
(I suspect the example is not minimal and it's coming from the `PolynomialTrendForecaster`.)

**Comment by eenticott-shell** [comment]
I have diagnosed where the error is coming from but I am unsure how to fix it. An X matrix is generated by `_sliding_window_transform`, this has 3 dimensions and is then passed to `fit` in the regressor, the regressor then rejects the X matrix as it has 3 dimensions. A naive fix is to do X.squeeze() to get rid of the extra dimension but this may break other functionality. Doing this does get the forecaster to fit successfully (though it raises an error as it has no summary attribute)

**Comment by fkiraly** [comment]
@eenticott-shell, it looks like the wrong recursive reducer is used.

There is one for *time series regressors* which I think produces 3D data to pass to an `sktime` time series regressor, the `RecursiveTimeSeriesRegressionForecaster`. But, in this case, I think the `RecursiveTabularRegressionForecaster` should be used, which produces 2D data to pass to a `sklearn` compatible regressor.

At some point, the wrong class must be selected, or a wrong argument passed to `make_reduction`.

**Comment by ArjunCodess** [comment]
hi, i'd like to work on this. reproduced on main: PolynomialTrendForecaster and make_reduction(LinearRegression()) both pass on the 2d tabular path. the dim-3 error only happens if you pass sklearn into RecursiveTimeSeriesRegressionForecaster yourself. can we close?
